devito/__init__.py

```diff
@@ -23,7 +23,11 @@
 from devito.data.allocators import *  # noqa
 from devito.logger import error, warning, info, set_log_level  # noqa
 from devito.mpi import MPI  # noqa
-from devito.checkpointing import DevitoCheckpoint, CheckpointOperator  # noqa
+from devito.checkpointing import pyrevolve  # noqa
+try:
+    from devito.checkpointing import DevitoCheckpoint, CheckpointOperator  # noqa
+except ImportError:
+    pass
```

Comment:
I think we can avoid a lot of 

```
if pyrevolve:
    ...
```

and a lot of `devito.DevitoCheckpoint`

by doing the following:

in `checkpointing.py` we could add:

```
class Noop(object):
    """
    Dummy replacement in case pyrevolve isn't available.
    """

    def __init__(self, *args, **kwargs):
        raise ImportError("Missing required dependency `pyrevolve`; cannot use checkpointing")

class NoopCheckpointOperator(Noop):
    pass

class NoopCheckpoint(Noop):
    pass
```

then here (line 30 of `__init__.py`) you import:

```
from devito.checkpointing import NoopCheckpoint as DevitoCheckpoint, NoopCheckpointOperator as CheckpointOperator
```

and then in theory we should just be good to go avoiding most of the changes we've made to examples and tests

(clearly in `test_checkpoint` we would still need the `skipif`)